MagazineLayout/LayoutCore/SectionModel.swift

```diff
@@ -503,17 +527,18 @@ struct SectionModel {
       }
     }
 
+    numberOfRows = rowIndex + 1
+
     if numberOfItems > 0 {
       // Apply bottom item inset now that we're done laying out items
       currentY += metrics.itemInsets.bottom
     }
 
     // Footer frame calculations
-    if rowIndex == indexOfFooterRow(), var footerModel = footerModel {
-      footerModel.originInSection = CGPoint(x: metrics.sectionInsets.left, y: currentY)
-      footerModel.size.width = metrics.width
-      updateFooterHeight(withMetricsFrom: footerModel)
-      self.footerModel = footerModel
+    if rowIndex == indexOfFooterRow(), let existingFooterModel = footerModel {
+      footerModel?.originInSection = CGPoint(x: metrics.sectionInsets.left, y: currentY)
+      footerModel?.size.width = metrics.width
+      updateFooterHeight(withMetricsFrom: existingFooterModel)
```

Comment:
Avoid bug due to mutating a copy of the footer model, not the footer model itself.